### ISSUE TITLE
Member Role Updates Not Immediately Being Reflected in Member State

### ISSUE THREAD

**Issue by Tiffinini** [Pycord-Development/pycord#1278]
### Summary

When changing a user's role, the user's list of roles does not reflect the changes until you explicitly retrieve the user again from the cache. It seems the state is not being updated for role assignment.

### Reproduction Steps

1. Create a command to edit a member's roles.
2. Add a role to the user.
3. Attempt to retrieve a list of the user's roles using the .roles method of the Member object, which will show the roles they had prior to having the new role added. It will not show the new role just added.

### Minimal Reproducible Code

```python
@slash_command(name="test-role-add", guild_ids=config.COMMAND_GUILD_ID_REGISTER_LIST)
@option("member", discord.Member, description="Member", required=True)
@option("role", discord.Role, description="Role", required=True)
async def test_add_roles(self, ctx: discord.ApplicationContext, member: discord.Member, role: discord.Role):
    print(f"Roles before: {member.roles}")  
        
    await member.add_roles(role)

    # This lists the roles as they were before the new role was added. The newly added role will not be shown.
    print(f"Roles after: {member.roles}")
```

### Expected Results

I expected the new role to be shown in the Member object's roles list. This is how the functionality was in at least beta 4.

### Actual Results

The list of roles was the same as before the new role was added. The only way to correct this is a call to the .get_member() method of the guild object to get a brand new instance of the member from the cache.

### Intents

All

### System Information

- Python v3.9.2-final
- py-cord v2.0.0-beta
    - py-cord pkg_resources: v2.0.0b7
- aiohttp v3.8.1
- system info: Windows 8.1 6.3.9600

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

It is worth noting that this behavior only occurs when the Member instance is passed from a command option. The following works absolutely fine, for example, when I get a member using the guild method:

```python
@slash_command(guild_ids=config.COMMAND_GUILD_ID_REGISTER_LIST)
async def no_input_role_test(self, ctx: discord.ApplicationContext):
    member = ctx.guild.get_member(164862518228025344)
    role = ctx.guild.get_role(728379815014563865)
        
    print(f"Roles before: {member.roles}")
        
    await member.add_roles(role)
        
    print(f"Roles after: {member.roles}")  # This will accurately show the updated roles for the user.   
```

It is worth noting that this was actually broke in [this specific commit to core.py](https://github.com/Pycord-Development/pycord/commit/409fe7a10dc5d2be6e92762aeb8793629b4006bd). L779 in that commit uses a directly-created Member object when previously it fetched a Member object from a guild cache using the util method get_or_fetch(), whose state received updates when things like roles changed.

Since I don't know the reason why this was changed or whether this was intended/desired behavior, I leave it to others to review.

**Comment by BobDotCom** [comment]
I believe the reason this is occurring is because options were changed to use the resolved data sent in the interaction, as opposed to fetching it (either from the API or cache). I'll look into this issue a bit more.

**Comment by plun1331** [comment]
This might be fixed in #1365
